include/ucl++.h

```diff
@@ -356,46 +424,136 @@ class Ucl final {
 		return out;
 	}
 
-	static Ucl parse (const std::string & in, std::string & err)
+	static Ucl parse (const std::string &in, std::string &err)
 	{
-		auto parser = ucl_parser_new (UCL_PARSER_DEFAULT);
+		return parse (in, std::map<std::string, std::string>(), err);
+	}
 
-		if (!ucl_parser_add_chunk (parser, (const unsigned char *)in.data (),
-				in.size ())) {
-			err.assign (ucl_parser_get_error (parser));
-			ucl_parser_free (parser);
+	static Ucl parse (const std::string &in, const std::map<std::string, std::string> &vars, std::string &err)
```

Comment:
In fact, you could use universal reference here instead of C++03 `const type &var`: `std::map<std::string, std::string> &&vars`